cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -267,7 +267,7 @@ func (p *BicepProvider) Deploy(ctx context.Context, pd *DeploymentPlan) (*Deploy
 		resourceManager := infra.NewAzureResourceManager(p.azCli)
 		progressDisplay := NewProvisioningProgressDisplay(resourceManager, p.console, bicepDeploymentData.Target)
 		// Make initial delay shorter to be more responsive in displaying initial progress
-		initialDelay := 3 * time.Second
+		initialDelay := 5 * time.Second
```

Comment:
We start this goroutine and wait for 3 seconds before the first poll, while also submitting a deployment in parallel. While running the tests, it is observed that the ARM deployment submission actually takes >3 seconds in most cases, which means that we would end up slowing down the responsiveness, since the first poll attempt would end up just not emitting any information. Thus, I increased the initial delay slightly as the minimal change here.

Reply:
Given the above change to stop earlier via an environment variable, did you still want this change?

Reply:
I do think it may be a better change, but wasn't data-driven here. Let me double check on performance numbers and see if this makes sense.

Reply:
50th percentile is below 3, so I'm reverting the change.